code/modules/admin/verbs/adminpm.dm

```diff
@@ -46,6 +46,25 @@
 		else		src << "<font color='red'>Error: Private-Message: Client not found. They may have lost connection, so try using an adminhelp!</font>"
 		return
 
+	if (src.handle_spam_prevention(msg,MUTE_ADMINHELP))
+		return
+
+	var/recieve_pm_type = "Player"
+	if(holder)
+		//mod PMs are maroon
+		//PMs sent from admins and mods display their rank
+		if(holder)
+			if(!C.holder && holder && holder.fakekey)
```

Comment:
We've checked for `holder` twice already. This condenses to `if (!C.holder && holder.fakekey)`